### DISCUSSION TITLE
Feature - Iteration Activity

### DISCUSSION THREAD

**Discussion by totpero** [elsa-workflows/elsa-core#2537] [Q&A] [answered]
Hi,

How can I Iterate trow activities while condition is true?
I want to come back to specific step if condition is not met like in image:

![image](https://user-images.githubusercontent.com/1710257/141653283-2b99e1ce-1f15-4c53-abe9-e2fc8596fad7.png)

**Comment by sfmskywalker** [comment, marked as answer]
Hi @totpero ,

You can use the same construct in Elsa by using the **If** activity and connect is **False** port to the activity you want to execute, and connect that activity back to the **If** activity. Just like you do in N8N.

**Reply by tomy2105** [reply]
I'm wondering since If in Elsa 2.0 (don't know if anything changed since) created scope and had done output, if connecting false back to something that executed that same If would do a lot of scopes and Dones executed at the end?

Wouldn't While be better here?

**Reply by sfmskywalker** [reply]
I just tried. The **If** scope is created only once and **Done** is executed only once.

Regardless, a **While** loop might indeed be better since it more closely models the looping construct.

But either approach should work well.
